integrations/tensorflow/compiler/TFSavedModelLowerGlobalTensors.cpp

```diff
@@ -73,9 +73,11 @@ static LogicalResult importTfSavedModelGlobalTensorsToIREEFlow(
              << "Multiple exported names for global tensor not supported yet";
     }
     symNameToFlowSymName[globalTensor.sym_name()] = flowSymName;
-    globalBuilder.create<IREE::Flow::VariableOp>(
+    auto variableOp = globalBuilder.create<IREE::Flow::VariableOp>(
         globalTensor.getLoc(), flowSymName, globalTensor.is_mutable(),
         globalTensor.type(), globalTensor.value());
+    SymbolTable::setSymbolVisibility(variableOp,
+                                     SymbolTable::Visibility::Private);
```

Comment:
This is only correct in the `exportedNames.empty()` case.

Reply:
IREE doesn't support exporting variables at all, so what's exportedNames supposed to be?

Reply:
exportedNames is a list of names that the SavedModel ABI requires making the variables available at.

If iree doesn't directly support a notion of an exported variable, we could synthesize get/set functions as per the semantics (that would satisfy the ABI as well).

But I thought that IREE did support exported variables? What exactly do you mean by exported in this context?

Reply:
What do _you_ mean by exported? ;)

The symbol visibility only impacts cross-module visibility, so:
```mlir
module @a {
  flow.variable @public_var
  flow.variable @private_var attributes { sym_visibility = "private" }
}
module @b {
  // @a::public_var is accessible
  // @a::private_var is not
}
```

It technically _is_ allowed prior to lowering into IREE, however once in the flow dialect transformation pipeline we need complete visibility of variable use and the only way to guarantee that is to treat variables as local to the module such that all uses can be discovered. If multiple input modules were being compiled we'd expect those to be combined into a single module or as you said use generated accessor functions to provide safe access. This is part of the ABI that gives us the ability to make variable accesses asynchronous: just as a normal function exported from an IREE module could have semaphores placed on it to stitch together work so too can the accessors, allowing cross-module async access in a way we can still model.

In C equivalence, all IREE variables are `static` and there is no way to do `extern static`, though you can provide functions that return those static variables to expose them beyond the compilation unit boundary.